src/react/useStream.ts

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from 'react';
+import { Observable } from 'rxjs';
+
+/**
+ * React hook to subscribe to a stream
+ *
+ * Each emit from the stream will make the component re-render with the new
+ * value. Initially the `initial` argument is returned, because an `Observable`
+ * has no guarantee for when the first emit will happen.
+ *
+ * This hook passes the `source$` argument as a dependency to `useEffect`, which
+ * means you will need to take care that it is referentially equal between each
+ * render (unless you want to resubscribe, of course).
+ *
+ * For using this hook with streams defined (created, piped, etc.) inside a
+ * component, you need to wrap the stream definition in `useMemo`:
+ * ```tsx
+ * const LatestPayload = ({ action }: { action: ActionWithPayload<any> }) => {
+ *   const payload$ = useMemo(() => action$.pipe(
+ *     ofType(action),
+ *     extractPayload()
+ *   ), [action]);
+ *   const payload = useStream(payload$, null);
+ *
+ *   return <p> Latest payload: { payload } </p>;
+ * }
+ * ```
+ *
+ * @param source$ Stream that provides the needed values
+ * @param initial Initial value to return
+ * @see useEffect
+ */
+export const useStream = <T, I>(source$: Observable<T>, initial: I): T | I => {
+  const [value, setValue] = useState<T | I>(initial);
```

Comment:
An observation from using a similar implementation of the hooks pattern a bit in one of our repos. When using the `useStream` hook think it's a bit annoying that you have to deal with the `initial` value within the view.

Normally when using `connect` to connect to a stream, the view doesn't render until the stream emits a value, which means that you don't have to deal with any intermediate value unless you choose to.

With this hooks pattern, you need to handle the `initial` value, although in practice what you actually want is to wait for the stream to emit or get the initial value from the stream.

Not sure if this can be done differently, but I think it has some implications for how we write views.